osu.Game/Screens/Play/HUD/BPMCounter.cs

```diff
@@ -26,6 +26,9 @@ public partial class BPMCounter : RollingCounter<double>, ISkinnableDrawable
         [Resolved]
         private IGameplayClock gameplayClock { get; set; } = null!;
 
+        [Resolved]
+        private Player player { get; set; } = null!;
```

Comment:
I think it should be possible to resolve `GameplayState` directly rather than go through player?